src/BigQuery/QueryResults.php

```diff
@@ -0,0 +1,213 @@
+<?php
+/**
+ * Copyright 2016 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace Google\Cloud\BigQuery;
+
+use Google\Cloud\BigQuery\Connection\ConnectionInterface;
+
+/**
+ * QueryResults represent the result of a BigQuery SQL query. Read more at the
+ * [Query Response API Documentation](https://cloud.google.com/bigquery/docs/reference/v2/jobs/query#response).
+ */
+class QueryResults
+{
+    /**
+     * @var ConnectionInterface $connection Represents a connection to BigQuery.
+     */
+    private $connection;
+
+    /**
+     * @var array The query result's metadata.
+     */
+    private $data;
+
+    /**
+     * @var array The query result's identity.
+     */
+    private $identity;
+
+    /**
+     * @var array The options to use when reloading query data.
+     */
+    private $reloadOptions;
+
+    /**
+     * @param ConnectionInterface $connection Represents a connection to
+     *        BigQuery.
+     * @param string $jobId The job's ID.
+     * @param string $projectId The project's ID.
+     * @param array $data The query result's metadata.
+     * @param array $reloadOptions The options to use when reloading query data.
+     */
+    public function __construct(ConnectionInterface $connection, $jobId, $projectId, array $data, array $reloadOptions)
+    {
+        $this->connection = $connection;
+        $this->data = $data;
+        $this->reloadOptions = $reloadOptions;
+        $this->identity = [
+            'jobId' => $jobId,
+            'projectId' => $projectId
+        ];
+    }
+
+    /**
+     * Retrieves the rows associated with the query and merges them together
+     * with the table's schema. It is recommended to check the completeness of
+     * the query before attempting to access rows.
+     *
+     * Example:
+     * ```
+     * $rows = $queryResults->getRows();
+     *
+     * foreach ($rows as $row) {
+     *     echo $row['name'];
+     * }
+     * ```
+     *
+     * @param array $options Configuration options.
+     * @return array
+     * @throws \Exception Thrown if the query has not yet completed.
+     */
+    public function getRows(array $options = [])
+    {
+        if (!$this->isComplete()) {
+            throw new \Exception('The query has not completed yet.');
```

Comment:
Also, is this an *exceptional* case where we'd want an exception to be thrown? Might be better to return an empty generator? What are the rules we're following on when exceptions are appropriate?